### PR TITLE
fix for #6151 @wdio/testingbot-service doesn't configure the hostname…

### LINKED ISSUE #6151: @wdio/testingbot-service doesn't configure the hostname/protocoll/port for tbTunnel option
**Environment (please complete the following information):**
 - **WebdriverIO version:** 6.9.1
 - **Mode:** WDIO Testrunner
 - **If WDIO Testrunner, running sync/async:** async
 - **Node.js version:** 10
 - **NPM version:** 6.14.8
 - **Browser name and version:** not relevant
 - **Platform name and version:** not relevant
 - **Additional wdio packages used (if applicable):** @wdio/testingbot-service (v6.8.2)

**Config of WebdriverIO**
https://gist.github.com/karfau/fbb7b5f2202c80c5c5fa61dab6f2f39d

**Describe the bug**
According to the documentation the only thing needed to run through the testinbot tunnel is to set tbTunnel option to `true`. But to actually make it work you also need to set the fields `hostname`, `protocol` and `port` correctly to actually make it work (see https://gist.github.com/karfau/fbb7b5f2202c80c5c5fa61dab6f2f39d#file-wdio-conf-js-L66-L68 in the provided config)

**To Reproduce**

```typescript
describe("browser support", () => {
  it("home page should show clickable button", async () => {
    await browser.url(`/`);
    const button = await $("button");
    expect(await button.waitForClickable({ timeout: 2000 })).toEqual(true);
  });
});
```

**Expected behavior**
The service already set's the three parameters (and `path`) to the correct values when `tbTunnel` option is not set or false. It should also set them correctly when it's set to true.

(Or) The (testingbot?) documentation could/should be more clear about HTTP vs HTTPS and the related ports (which combinations work) and when to use what.

### PR DESCRIPTION
**Pull request by testingbot**
## Proposed changes

This fixes #6151 - our latest change (e92dd59eb08eb0b585ebef2ebad72c26e4946004) was overwriting `tb:options` - this is now fixed with this PR.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-testingbot-service/src/@types/testingbot-service.d.ts [hunk 1]
```diff
@@ -19,5 +19,5 @@ interface TestingbotOptions {
      * Apply TestingBot Tunnel options (e.g. to change port number or logFile settings). See
      * [this list](https://github.com/testingbot/testingbot-tunnel-launcher) for more information.
      */
-    tbTunnelOpts?: object;
+    tbTunnelOpts?: any;
```
Comment (christian-bromann) [inline packages/wdio-testingbot-service/src/@types/testingbot-service.d.ts:22]: Could we get some actual type definition here?

File: packages/wdio-testingbot-service/src/launcher.ts [hunk 2]
```diff
@@ -19,23 +19,32 @@ export default class TestingBotLauncher {
...
+            for (const capability of capabilities) {
+                if (!capability['tb:options']) {
+                    capability['tb:options'] = {} as WebDriver.TestingbotCapabilities
+                }
 
-        tbOptions['tunnel-identifier'] = tbTunnelIdentifier
+                capability['tb:options']['tunnel-identifier'] = tbTunnelIdentifier
+            }
+        } else {
+            Object.keys(capabilities).forEach(browserName => {
```
Comment (christian-bromann) [inline packages/wdio-testingbot-service/src/launcher.ts:39]: let's do a for..of loop instead
```suggestion
            for (const browserName of Object.keys(capabilities)) {
```

File: packages/wdio-testingbot-service/testingbot-service.d.ts [hunk 3]
```diff
@@ -19,5 +19,5 @@ interface TestingbotOptions {
      * Apply TestingBot Tunnel options (e.g. to change port number or logFile settings). See
      * [this list](https://github.com/testingbot/testingbot-tunnel-launcher) for more information.
      */
-    tbTunnelOpts?: object;
+    tbTunnelOpts?: any;
```
Comment (christian-bromann) [inline packages/wdio-testingbot-service/testingbot-service.d.ts:22]: same as first comment

**Review by christian-bromann — CHANGES_REQUESTED** [review]

File: packages/wdio-testingbot-service/src/launcher.ts [hunk 4]
```diff
@@ -19,23 +19,32 @@ export default class TestingBotLauncher {
             return
         }
 
-        const tbTunnelIdentifier = `TB-tunnel-${Math.random().toString().slice(2)}`
+        const tbTunnelIdentifier = this.options.tbTunnelOpts && this.options.tbTunnelOpts.tunnelIdentifier ? this.options.tbTunnelOpts.tunnelIdentifier : `TB-tunnel-${Math.random().toString().slice(2)}`
```
Comment (christian-bromann) [inline packages/wdio-testingbot-service/src/launcher.ts:22]:
```suggestion
        const tbTunnelIdentifier = this.options.tbTunnelOpts?.tunnelIdentifier || `TB-tunnel-${Math.random().toString().slice(2)}`
```

File: packages/wdio-testingbot-service/src/launcher.ts [hunk 5]
```diff
@@ -19,23 +19,32 @@ export default class TestingBotLauncher {
...
+                if (!capability['tb:options']) {
+                    capability['tb:options'] = {} as WebDriver.TestingbotCapabilities
+                }
 
-        tbOptions['tunnel-identifier'] = tbTunnelIdentifier
+                capability['tb:options']['tunnel-identifier'] = tbTunnelIdentifier
+            }
+        } else {
+            for (const browserName of Object.keys(capabilities)) {
+                const capability = (capabilities as any)[browserName]
```
Comment (christian-bromann) [inline packages/wdio-testingbot-service/src/launcher.ts:40]:
```suggestion
                const capability = (capabilities as WebdriverIO.MultiRemoteOptions)[browserName]
```

**Review by christian-bromann — APPROVED** [review]
Awesome, thank you 👍

File: packages/wdio-testingbot-service/testingbot-service.d.ts [hunk 6]
```diff
@@ -9,6 +9,48 @@ declare module WebdriverIO {
     }
 }
 
+interface TunnelLauncherOptions {
+    // The TestingBot API key which you can get for free, listed in our member area
+    apiKey: string;
+ 
+    // The TestingBot API secret which you can get for free, listed in our member area
+    apiSecret: string;
```
Comment (karfau) [inline packages/wdio-testingbot-service/testingbot-service.d.ts:17]: Question: Are these options now mandatory? Is there no way to grab the ones from `user`/`key`? (Also it seems to work when not passing any options.)

File: packages/wdio-testingbot-service/src/@types/testingbot-service.d.ts [hunk 7]
```diff
@@ -19,5 +61,5 @@ interface TestingbotOptions {
      * Apply TestingBot Tunnel options (e.g. to change port number or logFile settings). See
      * [this list](https://github.com/testingbot/testingbot-tunnel-launcher) for more information.
      */
-    tbTunnelOpts?: object;
+    tbTunnelOpts?: TunnelLauncherOptions;
```
Comment (karfau) [inline packages/wdio-testingbot-service/src/@types/testingbot-service.d.ts:64]: Awesome! Thx for adding those types.

**Review by karfau — COMMENTED** [review]
Even though I have no idea why it fixes the issue I reported, I'm glad it does and thank you for the fast fix.

**Pull request merged**